app/v2/coinswap/internal/types/msgs.go

```diff
@@ -128,16 +136,19 @@ func (msg MsgAddLiquidity) Type() string { return "add_liquidity" }
 // ValidateBasic Implements Msg.
 func (msg MsgAddLiquidity) ValidateBasic() sdk.Error {
 	if !(msg.MaxToken.IsValid() && msg.MaxToken.IsPositive()) {
-		return sdk.ErrInvalidCoins("maxToken is invalid: " + msg.MaxToken.String())
+		return sdk.ErrInvalidCoins("max token is invalid: " + msg.MaxToken.String())
 	}
 	if msg.MaxToken.Denom == sdk.IrisAtto {
-		return sdk.ErrInvalidCoins("max token should be non-iris token: " + msg.MaxToken.String())
+		return sdk.ErrInvalidCoins("max token must be non-iris token")
+	}
+	if strings.HasPrefix(msg.MaxToken.Denom, FormatUniIdPrefix) {
+		return sdk.ErrInvalidCoins("max token must be non-liquidity token")
 	}
-	if msg.ExactIrisAmt.IsNil() || !msg.ExactIrisAmt.IsPositive() {
-		return ErrNotPositive("deposit amount provided is not positive")
+	if !msg.ExactIrisAmt.IsNil() && !msg.ExactIrisAmt.IsPositive() {
```

Comment:
Did not check ExactIrisAmt.IsNil()

Reply:
As discussed, we mark all `amounts` to be required